src/content_cao.cpp

```diff
@@ -613,56 +613,55 @@ bool GenericCAO::collideWithObjects()
 void GenericCAO::initialize(const std::string &data)
 {
 	infostream<<"GenericCAO: Got init data"<<std::endl;
+	refresh(data);
+
+	if (m_is_player) {
+		// Check if it's the current player
+		LocalPlayer *player = m_env->getLocalPlayer();
+		if (player && strcmp(player->getName(), m_name.c_str()) == 0) {
+			m_is_local_player = true;
+			m_is_visible = false;
+			player->setCAO(this);
+		}
+		m_env->addPlayerName(m_name.c_str());
+	}
+}
+
+void GenericCAO::refresh(const std::string &data)
```

Comment:
Maybe call this function `processMessages` or something like that? Refresh is very generic.